.dev/docker/env/Dockerfile

```diff
@@ -36,44 +34,55 @@ RUN apt-get update \
       libusb-1.0-0-dev \
       libvtk${VTK_VERSION}-dev \
       libvtk${VTK_VERSION}-qt-dev \
+      lsb-release \
       qtbase5-dev \
       software-properties-common \
+      wget \
+      xvfb \
+ && if [ "$USE_LATEST_CMAKE" = true ] ; then \
+      wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - | tee /etc/apt/trusted.gpg.d/kitware.gpg >/dev/null ; \
+      if wget -q --method=HEAD "https://apt.kitware.com/ubuntu/dists/$(lsb_release -cs)/Release"; then \
+        apt-add-repository "deb https://apt.kitware.com/ubuntu/ $(lsb_release -cs) main" ; \
+      else \
+        apt-add-repository 'deb https://apt.kitware.com/ubuntu/ focal main' ; \
+      fi \
```

Comment:
1. Do we need the focal fallback? Shouldn't 21.04 have the same or newer CMake?
2. If no, can we separate out the URL into a variable?

Reply:
1) 21.04 does only provide CMake 3.18 (see [here](https://packages.ubuntu.com/search?suite=all&searchon=names&keywords=cmake))

2) I don't think this increases the readability when we have

```sh
      cmake_ubuntu_release="focal" \
      cmake_repo_url="https://apt.kitware.com/ubuntu/" \
      if wget -q --method=HEAD "$(cmake_repo_url)dists/$(lsb_release -cs)/Release"; then \
        cmake_ubuntu_release="$(lsb_release -cs)" ; \
      fi \
      apt-add-repository "deb $(cmake_repo_url) $(cmake_ubuntu_release) main" ; \
```

Reply:
Regarding Point 2, it's a lost cause.

Regarding the use of 20.04 binaries in 21.04, is it a safe idea?

Apart from this, PR LGTM

Reply:
> Regarding the use of 20.04 binaries in 21.04, is it a safe idea?

There is no guarantee that it will work. The 18.04 packages do not work on 21.04 for example, because it does not find some dependency. I have tested the 20.04 packages on 20.10 and 21.04, but it may be that the 20.04 packages no longer work on 21.10.

Reply:
Wouldn't it be easier to download and install to ensure compatibility?

Reply:
You mean to download e.g. `cmake-3.20.3-linux-x86_64.sh` and then install it? Don't think so, as I need somehow than the information, which is the latest version.

In general, I think that we don't need to worry about future versions of Ubuntu just yet, since incompatibilities can occur even so (could be that there are packages missing or other required).

Reply:
git clone + `git tag -l | grep -v rc | sort -V | tail -n 1` ? get the latest non release candidate tag to switch and then `./bootstrap && make -j2 install` or similar?

Reply:
It is more complicated as CMake also have some dependency, when you want to compile it (e.g. OpenSSL headers). As long as we don't have any issues with the precompiled deb files I think we should use them.

Reply:
I will simplify the code a bit after [these](gitlab.kitware.com/cmake/cmake/-/issues/22245) changes -just waiting that the issue getting closed there ;-)